### PR TITLE
Feature alerting mail support html format #586

### PR DESCRIPTION
**Pull request by dankyalo599**
### Description
Feature alerting mail support html format #586

### Issues Resolved
To make the MessageContent class support HTML format for email notifications,I  modified the buildMessageWithTitle method to take into account the htmlDescription property.
With this change, the method will return the htmlDescription property if it is not null, otherwise it will return the concatenation of the title and text description as before.
Furthermore , I modified the init block to check if the htmlDescription property is not null, since it's no longer required.

### Check List
- [X ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/dashboards-notifications-test-and-build-workflow.yml [hunk 1]
```diff
@@ -153,7 +153,7 @@ jobs:
         uses: actions/cache@v2
         with:
           path: ${{ matrix.cypress_cache_folder }}
-          key: cypress-cache-v2-${{ runner.os }}-${{ hashFiles('**/package.json') }}
+          key: cypress-cache-v2-${{ runner.os }}-${{ hashFiles('OpenSearch-Dashboards/plugins/dashboards-notifications/package.json') }}
```
Comment (lezzago) [inline .github/workflows/dashboards-notifications-test-and-build-workflow.yml:156]: why is this needed?

File: notifications/core-spi/src/main/kotlin/org/opensearch/notifications/spi/model/destination/DestinationType.kt [hunk 2]
```diff
@@ -8,5 +8,5 @@ package org.opensearch.notifications.spi.model.destination
  * Supported notification destinations
  */
 enum class DestinationType {
-    CHIME, SLACK, CUSTOM_WEBHOOK, SMTP, SES, SNS
+    CHIME, SLACK, CUSTOM_WEBHOOK, SMTP, SES, SNS, TELEGRAM
```
Comment (lezzago) [inline notifications/core-spi/src/main/kotlin/org/opensearch/notifications/spi/model/destination/DestinationType.kt:11]: remove telegram from this PR as its part of another pr

**Comment by lezzago** [comment]
Please provide screen shots of the html support working.

**Comment by gaobinlong** [comment]
Repeated PR, close this.
